client/ayon_core/hosts/resolve/plugins/load/load_media.py

```diff
@@ -0,0 +1,365 @@
+            project_name,
+            version["parent"],
+            fields=["name"]
+        ) or {}
+
+        # set clip colour
+        if version.get("name") == last_version.get("name"):
+            return cls.clip_color_last
+        else:
+            return cls.clip_color
+
+    def _set_metadata(self, media_pool_item, context: dict):
+        """Set Media Pool Item Clip Properties"""
+
+        # Set the timecode for the loaded clip if Resolve doesn't parse it
+        # correctly from the input. An image sequence will have timecode
+        # parsed from its frame range, we will want to preserve that.
+        # TODO: Setting the Start TC breaks existing clips on update
+        #   See: https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=197327
+        #   Once that's resolved we should enable this
+        # start_tc = media_pool_item.GetClipProperty("Start TC")
+        # if start_tc == "00:00:00:00":
+        #     from openpype.pipeline.editorial import frames_to_timecode
+        #     # Assume no timecode was detected from the source media
+        #
+        #     fps = float(media_pool_item.GetClipProperty("FPS"))
+        #     handle_start = context["version"]["data"].get("handleStart", 0)
+        #     frame_start = context["version"]["data"].get("frameStart", 0)
+        #     frame_start_handle = frame_start - handle_start
+        #     timecode = frames_to_timecode(frame_start_handle, fps)
+        #
+        #     if timecode != start_tc:
+        #         media_pool_item.SetClipProperty("Start TC", timecode)
+
+        # Set more clip metadata based on the loaded clip's context
+        metadata = {
+            "Clip Name": "{asset[name]} {subset[name]} "
+                         "v{version[name]:03d} ({representation[name]})",
+            "Shot": "{asset[name]}",
+            "Take": "{subset[name]} v{version[name]:03d}",
+            "Comments": "{version[data][comment]}"
+        }
+        for clip_property, value in metadata.items():
+            media_pool_item.SetClipProperty(clip_property,
+                                            value.format_map(context))
+
+    def _get_filepath(self, representation: dict) -> Union[str, dict]:
+
+        is_sequence = bool(representation["context"].get("frame"))
+        if not is_sequence:
+            return get_representation_path(representation)
+
+        context = get_representation_context(representation)
+        version = context["version"]
+
+        # Get the start and end frame of the image sequence, incl. handles
+        frame_start = version["data"].get("frameStart", 0)
+        frame_end = version["data"].get("frameEnd", 0)
+        handle_start = version["data"].get("handleStart", 0)
+        handle_end = version["data"].get("handleEnd", 0)
+        frame_start_handle = frame_start - handle_start
+        frame_end_handle = frame_end + handle_end
+        padding = len(representation["context"].get("frame"))
```

Comment:
I've heard of cases where a particular representation might have differing frame ranges than the version data (e.g. slates added, etc.) and as such this MIGHT be incorrect for some edge cases. We might need to at some point store those ranges in `representation["data"]` so we can reliably get frame ranges per representation instead.

Reply:
This is good point for the new Representation model @antirotor is working on.